### DISCUSSION TITLE
Auto-import vue SFC from custom dir

### DISCUSSION THREAD

**Discussion by jemdiggity** [nuxt/nuxt#26323] [Questions] [answered]
I'm trying to setup a monorepo for web/mobile with ionic/capacitor/nuxt. My dir structure is as suggested [here](https://ionic.nuxtjs.org/cookbook/web-and-device)
```
- apps
  - nuxt-web
    - ...
    - nuxt.config.ts
  - nuxt-device
    - ...
    - nuxt.config.ts
- packages
  - core
    - components
    - composables
    - ...
```

Auto-imports is working fine for composables, doesn't work for SFC components in `packages/core/components`.
My nuxt.config.ts has
```
  imports: {
    dirs: ["../packages/core/components", "../packages/core/composables"],
  },
```

I get a `Failed to resolve component: MyComponent` warning.
Should this work? TIA.

**Comment by xjccc** [comment, marked as answer]
Your scene, let packages/core be a layers? In layers `components` and `composable` is auto-import in your project.
[layers](https://nuxt.com/docs/getting-started/layers)

**Reply by jemdiggity** [reply]
Thanks for the tip. I setup a CommonUi app and the App’s nuxt config extends from it.
